docs/api/v1.md

```diff
@@ -380,49 +382,82 @@ Responses:
 }
 ```
 
-### Function
+### SourceFunction
 ```
 {
    address {integer}: Memory address of the function
    name {string}: Name of the function
    sourcePath {string}: Full path to the source code
    lineNumber {integer}: Line number of the function definition
    lineCount {integer}: Length of function in source code lines
-   parameters {Variable[]}: Array of parameter variables
-   localVariables {Variable[]): Array of all local variables defined in function
+   parameters {SourceVariable[]}: Array of parameter variables
+   localVariables {SourceVariable[]): Array of all local variables defined in function
 }
 ```
 
 ### DebugInfo
 ```
 {
   id {DebugId}
-  attachedProcess {[Process](#process)}
+  attachedProcess {Process}
   sourcePath {string}: Path to root directory of source code
 }
 ```
 
 ### Breakpoint
 ```
 {
-  function {[Function](#Function)}: The function that this is breakpoint is associated with.
+  sFunction {SourceFunction}: The function that this is breakpoint is associated with.
   metadata {string}: Any additional info associated with this breakpoint.
 }
 ```
 
-### Variable
+### SourceVariable
 ```
 {
   id {var id}: Unique identifier of that variable.
   name {string}: Name of the variable as it appears in the function.
-  sType {SourceType}: Type of the variable as it is defined in the source code.
+  sValueIndex {number}: Index to the variable in the SourceValue[] as it is defined in the source code.
   address {integer}: Memory address
 }
 ```
 
-### SourceType
-TODO
+### SourceValue
+```
+{
+  sTypeIndex {integer}: Index into SourceTypeDefinition array of the type that this variable is.
+  data {boolean | integer | double | null}: Only used if sTypeIndex points to a type of `kind == "primitive"`. Contains the primitive data.
+  target {integer}: Only used if sTypeIndex points to a type of `kind == "pointer"`. Contains the index into the SourceType array that this points to.
+  fields { name {string}: integer}: Only used if sTypeIndex points to a type of `kind == "struct"`. Map of indexes into the SourceType array that contain definitions for each field of a given name. 
+}
```

Comment:
Combine `data`, `target`, and `fields` fields into just `data`, or alternatively put them in different types combined through `|` or inheritance. (Inheritance could put `sTypeIndex` in a base interface.)